### PR TITLE
[UNR-1207] Documentation - SpatialGDK settings

### PR DESCRIPTION
**Pull request by improbable-valentyn**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
Describe your changes here.

#### Release note
REQUIRED: Enter a simple message for this commit which will go in the release notes. Prefix with "Bugfix:" or "Feature:".
(Documentation changes are exempt)

#### Tests
How did you test these changes prior to submitting this pull request?
What automated tests are included in this PR?

STRONGLY SUGGESTED: How can this be verified by QA?

#### Documentation
How is this documented (for example: release note, upgrade guide, feature page, in-code documentation)?

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

**Comment by improbable-valentyn** [comment]
TODO: rebase and repoint against 0.4.0-rc

**Comment by eddyrainy** [comment]
The doc draft for GDK Editor Settings is completed and here's the screenshot of the Improbadoc. For each property name and tooltip, please use the first sentence of each description excluding the "for example" part.

image

image

**Comment by eddyrainy** [comment]
The draft for both panels are completed. Here's the screenshot I got:

SpatialOS Editor Settings:
image

SpatialOS Runtime Settings:
image
image

**Comment by eddyrainy** [comment]
## Designing embedded assistance using the **progressive disclosure** method

Before you begin to update UI elements for SpaitalOS Editor/Runtime settings, read the **background**  section to get a little understanding of embedded assistance and progressive disclosure. 

After that, check **Updating embedded assistance of SpatialOS Editor/Runtime settings** to apply the changes.

### Background

**Embedded assistance**: Information is just where users need it and just when they need it - ideally the interface with its associated text is so clear that users don't need additional assistance.

**Embedded assistance** encompasses all text that users encounter in all types of products, including all text in user interfaces, but also even command names; users use all terms to form a mental model about what to do.

**Progressive disclosure** is about revealing information in an ordered manner, providing a clear trail, connecting to the big picture, and designing around the ideal information experience.

By adopting the progressive disclosure method, we design levels of embedded assistance, i.e. Map the embedded assistance elements you have or want to use and the content for each, for example, 
image

Where:
- Level 1: programmatic assistance
- Level 2: UI labels, messages
- Level 3: Inline UI text
- Level 4: Control-level assistance, hover help
- Level 5: Panel-level assistance

### Updating embedded assistance of SpatialOS Editor/Runtime settings

**Level 1: programmatic assistance - we make decisions for users**
- Check box selected or deselected by default; default values provided; invalid fields are greyed out.
- Ensure that what's shown above in the user interface is consistent with the documentation. 

**Level 2: label and strings - we provide clear, descriptive text.**
- Apply sentence-style capitalization to UI labels and strings to improve consistency, implementation, and readability.
- Implement my edits to UI labels and strings. For a complete list, see the Gdocs. Ensure the property names in the user interface are consistent with the documentation. 
image

**Level 3: Instructional text in UI - we show additional instruction to provide context, orientation, warnings, examples.**

In the following screenshot, change the words (_figure 1_) to 
- For **SpatialOS Editor Settings**: Configure your local deployment, and schema and snapshot generation.
- For **SpatialOS Runtime Settings**: Configure your game at runtime.

Meanwhile, change the words  (_figure 2_) to 
- For **SpatialOS Editor Settings**: The settings are saved in the DefaultSpatialGDKEditorSettings.ini file. To learn more, see the Reference section of the [documentation](https://docs.improbable.io/unreal/).
- For **SpatialOS Runtime Settings**: The settings are saved in the DefaultSpatialGDKSettings.ini file. To learn more, see the Reference section of the [documentation](https://docs.improbable.io/unreal/).
image

**Level 4: Control-level text: hover help, pop up messages**
- A short descriptive phrase without punctuation, esp. ending punctuation (such as a period).
- General 50 or fewer words, leave expansion space for translation.
- Try not to obscure other UI controls when the tooltip appears.
- Apply my edits to the tooltips. See the Gdocs. 

**Level 5: Panel level assistance**
- I don't think that we need it now, for example, the following one:
image

File: SpatialGDK/Documentation/toc.md [hunk 1]
```diff
@@ -34,6 +34,8 @@
     - [Directory structure]({{urlRoot}}/content/directory-structure)
     - [Gameplay Ability System]({{urlRoot}}/content/ability-system)
     - [Non-Unreal server-worker types]({{urlRoot}}/content/non-unreal-server-worker-types)
+    - [SpatialOS Editor Settings]({{urlRoot}}/content/editor-settings)
```
Comment (ElleEss) [inline SpatialGDK/Documentation/toc.md]: Suggest go with the Toolbar page under the title "Unreal Editor Interface" which matches the Unreal docs section on this: https://docs.unrealengine.com/en-US/Engine/UI

**Comment by ernopp** [comment]
Discussed with Laura, this PR is out of date and will be replaced by a new one based on the Gdoc edits (see DOC-1161)

**Pull request closed without merging**